### DISCUSSION TITLE
Use default value with null inputs

### DISCUSSION THREAD

**Discussion by gregoirechauvet** [MichalLytek/type-graphql#1027] [Q&A] [answered]
On a boolean input field that has a default value, a GraphQL client can send:
- Omit the field (i.e. `undefined` in JavaScript)
- `null`
- A boolean

The default value will only be used in the `undefined` case. Whereas in the `null` case, `null` is actually what is received in the resolver.
From my understanding, it's the expected behavior from the GraphQL specification. `null` is considered a value.
(even though it's quite inconvenient in my opinion :stuck_out_tongue:)

I would like to always have a boolean here, in all cases. My example here is with a boolean, but it works on all input types.

How the input is declared:
```ts
@InputType()
class MyInput {
  @Field({ defaultValue: false })
  myProperty: boolean;
}
```

One solution is to fix the type and add  `| null`, and handle this case in the resolver. But it's quite cumbersome.
```diff
@InputType()
class MyInput {
  @Field({ defaultValue: false })
--  myProperty: boolean;
++  myProperty: boolean | null;
}
```

I'm looking for a simpler solution. I tried to use the `@UseMiddleware()` decorator, but it seems they are not working on inputs, only on `@ObjectType`. Then I tried to wrap the `@Field` decorator, but I'm not able to patch the values either.

Would you have a solution, or a lead to investigate? What would be the best way to do it?

----

I think the TypeGraphQL documentation is not very clear on this topic. There are no hints, not even with the type system that `null` can be received, while it's a valid input.

**Comment by MichalLytek** [comment, marked as answer]
I think your issue has been solved in #806, so when you provide default value, it will no more emit the field as nullable, so `null` value won't be allowed.

**Reply by gregoirechauvet** [reply]
Hum... but what's the point of a default value if the field is non-nullable? :thinking: 
As far as I know, nullable means the field can be omitted, or `null` can be sent, right? Otherwise, the client is required to send a value. Am I missing something?

**Reply by MichalLytek** [reply]
Read the discussion in the original issue, everything about default value and null is described there.

**Reply by gregoirechauvet** [reply]
I did but I'm still confused. But I can try the change and play with it myself.
I'll try with the version from `master` has the change hasn't been released yet. Thanks for your answers!

**Reply by gregoirechauvet** [reply]
It works perfectly fine, my understanding of the GraphQL specification wasn't correct. Awesome!
Any new release planned then? :innocent:

**Reply by MichalLytek** [reply]
Yes, I need to finish a few things and then v1.2 will be ready!
